staging/src/k8s.io/csi-translation-lib/plugins/gce_pd.go

```diff
@@ -73,20 +73,25 @@ func translateAllowedTopologies(terms []v1.TopologySelectorTerm) ([]v1.TopologyS
 	newTopologies := []v1.TopologySelectorTerm{}
 	for _, term := range terms {
 		newTerm := v1.TopologySelectorTerm{}
+		zoneFound := false
 		for _, exp := range term.MatchLabelExpressions {
 			var newExp v1.TopologySelectorLabelRequirement
-			if exp.Key == v1.LabelZoneFailureDomain {
+			if !zoneFound && (exp.Key == v1.DeprecatedLabelZoneFailureDomain || exp.Key == v1.LabelZoneFailureDomain) {
 				newExp = v1.TopologySelectorLabelRequirement{
 					Key:    GCEPDTopologyKey,
 					Values: exp.Values,
 				}
+
+				zoneFound = true
```

Comment:
what does this `zoneFound` thing do? seems to short-circuit the loop in a way and ignore any further zones. Why would we want this?

Reply:
It's used in the if statement on line 79. Without it, we would append both beta and GA topology selector to `newTerm.MatchLabelExpression`. I think what we want is to only append the first one we see.